src/include/planner/operator/extend/logical_recursive_extend.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include "binder/query/reading_clause/bound_gds_call.h"
+#include "planner/operator/logical_operator.h"
+
+namespace kuzu {
+namespace planner {
+
+class LogicalRecursiveExtend : public LogicalOperator {
+    static constexpr LogicalOperatorType operatorType_ = LogicalOperatorType::RECURSIVE_EXTEND;
+
+public:
+    LogicalRecursiveExtend(binder::BoundGDSCallInfo info, common::table_id_set_t nbrTableIDSet,
+        std::shared_ptr<LogicalOperator> nodeMaskRoot)
+        : LogicalOperator{operatorType_}, info{std::move(info)},
+          nbrTableIDSet{std::move(nbrTableIDSet)}, limitNum{common::INVALID_LIMIT},
+          nodeMaskRoot{std::move(nodeMaskRoot)} {}
+
+    void computeFlatSchema() override;
+    void computeFactorizedSchema() override;
+
+    const binder::BoundGDSCallInfo& getInfo() const { return info; }
+    binder::BoundGDSCallInfo& getInfoUnsafe() { return info; }
+
+    void setNbrTableIDSet(common::table_id_set_t set) { nbrTableIDSet = std::move(set); }
+    bool hasNbrTableIDSet() const { return !nbrTableIDSet.empty(); }
+    common::table_id_set_t getNbrTableIDSet() const { return nbrTableIDSet; }
+
+    void setLimitNum(common::offset_t num) { limitNum = num; }
```

Comment:
I don't think you handled `LogicalRecursiveExtend` in `LimitPushDownOptimizer`. Can you revisit that?